### PR TITLE
devx: add notes on buildkite-job-dispatcher

### PR DESCRIPTION
**Pull request by bobheadxi**
As pointed out in the [devx on-call proposal](https://docs.google.com/document/d/1KejbYYumVeu0eqzY4cLyCp6HHDzQbC7d9keFUBJeioU/edit#heading=h.qrgtkrm14j3l), this area is a bit poorly surfaced from responsibilities. This PR adds a high-level overview of how it works and where to go to learn more

https://deploy-preview-4226--sourcegraph-handbook.netlify.app/departments/engineering/dev/tools/infrastructure/ci/#buildkite-job-dispatcher

### CODE REVIEW

File: content/departments/engineering/dev/process/incidents/playbooks/ci.md [hunk 1]
```diff
@@ -242,4 +242,4 @@ In order to handle problems with the CI, the following elements are necessary:
...
 3. Check [dispatcher logs](https://console.cloud.google.com/logs/query;lfeCustomFields=jsonPayload%252FexpectDispatch;query=resource.type%3D%22k8s_container%22%0Aresource.labels.project_id%3D%22sourcegraph-ci%22%0Aresource.labels.location%3D%22us-central1-c%22%0Aresource.labels.cluster_name%3D%22default-buildkite%22%0Aresource.labels.namespace_name%3D%22buildkite%22%0Aresource.labels.container_name%3D%22buildkite-job-dispatcher%22;summaryFields=jsonPayload%252FrunID,jsonPayload%252Fmsg,jsonPayload%252F%2522metric.agentsDispatched%2522,jsonPayload%252F%2522metric.freeAgents%2522,jsonPayload%252F%2522metric.scheduledJobs%2522,jsonPayload%252F%2522metric.totalAgents%2522:false:18:end:true;timeRange=PT30M?project=sourcegraph-ci) for details
 
-For more details, see the source: [buildkite-job-dispatcher](https://sourcegraph.com/github.com/sourcegraph/infrastructure/-/tree/docker-images/buildkite-job-dispatcher)
+For more details, see the source: [buildkite-job-dispatcher](https://github.com/sourcegraph/infrastructure/blob/main/buildkite/kubernetes/buildkite-job-dispatcher/README.md)
```
Comment (bobheadxi) [inline content/departments/engineering/dev/process/incidents/playbooks/ci.md:245]: slowly removing sourcegraph.com links to private repos. could probably set up a BC for this at some point

File: content/departments/engineering/dev/tools/infrastructure/ci/index.md [hunk 2]
```diff
@@ -26,9 +26,84 @@ We maintain a shared fleet of Buildkite agents for continuous integration across
...
+- [Pipeline](https://buildkite.com/sourcegraph/buildkite-git-references)
+
+Relevant runbooks:
+
+- [Replacing agents](../../../process/incidents/playbooks/ci.md#replacing-agents)
+- [Agent availability issues](../../../process/incidents/playbooks/ci.md#agent-availability-issues)
+
+A diagram overview of how the `buildkite-job-dispatcher` works (diagram adapted from [here](https://bobheadxi.dev/stateless-ci/#dynamic-kubernetes-jobs)):
+
+```mermaid
```
Comment (bobheadxi) [inline content/departments/engineering/dev/tools/infrastructure/ci/index.md:58]: couldn't get the diagram rendering :( asking for help here: https://sourcegraph.slack.com/archives/C02PSJF0QLU/p1657315582220539

Comment (bobheadxi) [reply content/departments/engineering/dev/tools/infrastructure/ci/index.md:58]: worst case I'll add a link to view this in GitHub where it renders: [Update index.md](https://github.com/sourcegraph/handbook/pull/4226/commits/ba16a79a2ac5f30aa816ceda4006a4242ea93bdf)

File: content/departments/engineering/dev/tools/infrastructure/ci/index.md [hunk 3]
```diff
@@ -26,9 +26,84 @@ We maintain a shared fleet of Buildkite agents for continuous integration across
...
+          end
+        end
+      end
+      deactivate ba
+
+      k8s->>k8s: Clean up completed Jobs
+
+      deactivate k8s
+    end
+```
```
Comment (bobheadxi) [inline content/departments/engineering/dev/tools/infrastructure/ci/index.md:109]:
```suggestion
```

> NOTE: If this diagram fails to render, [read this page in GitHub instead](https://github.com/sourcegraph/handbook/blob/main/content/departments/engineering/dev/tools/infrastructure/ci/index.md#buildkite-job-dispatcher)
```

File: content/departments/engineering/dev/tools/infrastructure/ci/index.md [hunk 4]
```diff
@@ -26,9 +26,86 @@ We maintain a shared fleet of Buildkite agents for continuous integration across
...
+
+Relevant runbooks:
+
+- [Replacing agents](../../../process/incidents/playbooks/ci.md#replacing-agents)
+- [Agent availability issues](../../../process/incidents/playbooks/ci.md#agent-availability-issues)
+
+A diagram overview of how the `buildkite-job-dispatcher` works (diagram adapted from [here](https://bobheadxi.dev/stateless-ci/#dynamic-kubernetes-jobs)):
+
+```mermaid
+sequenceDiagram
```
Comment (jhchabran) [inline content/departments/engineering/dev/tools/infrastructure/ci/index.md:59]: 😍

**Review by jhchabran — APPROVED** [review]
I hope we can get the diagram to work inline, that'll help so much 🙏 

Thanks for taking on this 💪

**Pull request merged**